### ISSUE TITLE
Feature Request: Legends for all plots

### ISSUE THREAD

**Issue by timkofu** [facebook/prophet#1838]
Maybe I'm missing a setting, but there currently seems to be no inbuilt in way to enable legends for the plot functions. Would it be possible to add this feature? Many thanks.

**Comment by bletham** [comment]
That is correct. The components plot only have 1 line so there isn't need for a legend there, but I can see why one might want to add a legend to the `m.plot()` plot. A flag "include_legend" to `m.plot` would be a nice addition.

In the meantime, you can do this directly in matplotlib like so:
```Python
fig = m.plot(fcst)  # This returns a matplotlib Figure
ax = fig.gca()  # Now you have the matplotlib Axis
ax.legend([....])  # Put whatever legend you want here.
```
Once you have `ax`, you can make arbitrary changes to the plot using matplotlib.

**Comment by vshargin** [comment]
I can help adding this to the Python package.
